### DISCUSSION TITLE
Transition to cgroups v2

### DISCUSSION THREAD

**Discussion by JaimeMagiera** [okd-project/okd#611] [General]
This thread is to collate all the information we have on handling FCOS's transition to cgroups v2, including the game plan for OKD to support it.  

https://github.com/coreos/fedora-coreos-tracker/issues/292#issuecomment-819791190

**Comment by vrutkovs** [comment]
My experimental PR to enable cgroupsv2 - https://github.com/openshift/okd-machine-os/pull/88

Current showstopper is https://bugzilla.redhat.com/show_bug.cgi?id=1949438 - builds are broken in cgroupsv2. Seems builder repo needs a fresh runc included.

It would be great to figure out what `psi/cgroup_no_v1` kernel args do, as I blindly copied those from some OCP PR, so not sure if these are critical

**Reply by SriRamanujam** [reply]
The `psi=1` kernel argument will enable pressure stall information tracking. 

Enabling it will create some files under `/proc/pressure/` with pressure stall information for cpu, memory, and i/o. https://www.kernel.org/doc/html/latest/accounting/psi.html

`cgroup_no_v1=all` apparently blacklists all cgroup v1 controllers, but leaves them available in v2. Quoting from the kernel docs:
```
        cgroup_no_v1=   [KNL] Disable cgroup controllers and named hierarchies in v1
                        Format: { { controller | "all" | "named" }
                                  [,{ controller | "all" | "named" }...] }
                        Like cgroup_disable, but only applies to cgroup v1;
                        the blacklisted controllers remain available in cgroup2.
                        "all" blacklists all controllers and "named" disables
                        named mounts. Specifying both "all" and "named" disables
                        all v1 hierarchies.
```
ref. https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html

**Reply by vrutkovs** [reply]
>The psi=1 kernel argument will enable pressure stall information tracking.

That looks useful, I'll add this param to MCO's allowlist.

We probably should start without cgroup_no_v1 for now and enable it once we confirm cgroupsv2 works

**Comment by vrutkovs** [comment]
Discussed on today's OKD WG meeting. The plan is:
* [Automatically enable cgroupsv2 by default for new installs (load it in bootstrap node via okd-machine-os manifest)](https://github.com/openshift/okd-machine-os/pull/227)
* Gather feedback, fix bugs
* Enable it for everyone by moving to okd-machine-os manifests applied on upgrade
